### ISSUE TITLE
curly_braces_in_flow_control_structures lints unnecessarily most of the time

### ISSUE THREAD

**Issue by creativecreatorormaybenot** [dart-lang/linter#1763]
**Describe the issue**

[The documentation states](https://dart-lang.github.io/linter/lints/curly_braces_in_flow_control_structures.html) that `curly_braces_in_flow_control_structures` is supposed to prevent the *dangling else* problem, but the linter also marks cases that are completely unrelated to this and unhelpful in my opinion.

**To Reproduce**

```dart
for (...) doStuff();
```

See also: https://github.com/dart-lang/pedantic/issues/30

**Expected behavior**

I think that the lint should only be triggered when there actually is a scenario with multiple `if`s that could be meant by the `else`.

**Comment by davidmorgan** [comment]
This is what the style guide requires :)

https://dart.dev/guides/language/effective-dart/style#do-use-curly-braces-for-all-flow-control-structures

That's a good reason to lint it :) ... and means that if you do think it should be changed then you'd have to ask for a style guide change, first. You're welcome to start that conversation--it's always reasonable to ask--but I suspect it's unlikely to get very far. Sorry!

**Comment by creativecreatorormaybenot** [comment]
@davidmorgan What about `for`-loops? I do not understand why the curly braces are required there. I could also not find that in the style guide.

**Comment by davidmorgan** [comment]
It's covered by that same rule, although it's not mentioned explicitly. The lint is supposed to exactly implement that rule of the style guide. If it doesn't, that's a bug :)

**Comment by creativecreatorormaybenot** [comment]
@davidmorgan Where would I ask for a style guide change?

**Comment by pq** [comment]
+1.  I think the issue needs to get resolved in the style guide.  If there's a change there we can adjust this rule to follow suit.  Thanks for kicking off the conversation @creativecreatorormaybenot!

**Comment by creativecreatorormaybenot** [comment]
@pq I opened an issue: https://github.com/dart-lang/site-www/issues/1994

However, if that is adjusted, the lint rule needs to be adjusted as well.

**Comment by pq** [comment]
Great.  Will track.  Thanks!
